Logs Submission by Cybrilla for Investment New Folio and existing folio

swapnil-cyb opened the pull request.

sourabhsaini11 commented:
@swapnil-cyb, kindly re upload logs in correct folder/file format, here's a ref [link](https://github.com/ONDC-Official/verification-logs/tree/main/FIS14)

swapnil-cyb commented:
Hi @sourabhsaini11 re uploaded the logs.
Kindly review.

sourabhsaini11 commented:
@swapnil-cyb, please provide only context & message as part of JSON payload, remove other fields/attributes send.

92shreyansh commented:
@sourabhsaini11 @anuj-ondc Please pick this up and in case we have utility it wouldbe great to share that

swapnil-cyb commented:
@sourabhsaini11 
As mentioned above fulfillments array can't be empty
we are getting this array as empty on onSelect response which is not supposed to be empty as per https://ondc-official.github.io/ONDC-FIS-Specifications/?branch=release-FIS14-2.0.0&flowId=Lumpsum+-+New+Folio&tabId=%23flow&callId=%23On+Select#On%20Select.

Can you please fix this along with other issues in on_Action call as mentioned in previous comment by @arcybrilla  so that we can close this.

swapnil-cyb commented:
Hi @sourabhsaini11 re uploaded the logs as per suggestions.
We fixed all of the reported issues except one below

```
init ->
quote is missing
```

Reason as per specification quote is not part of init flow.

Also, added redemption, payment retry flows.
Kindly review.

Srimaan-Guntoju commented:
@92shreyansh @sourabhsaini11 We will be doing the Investor KYC outside of ONDC flows before they place the transaction.

sourabhsaini11 commented:
@swapnil-cyb, excluding /on_action call issues
## Flow Lumpsum_Payment_Retry
## common_issue
additional calls **select_2**, **on_select_2**, **13_on_status**, **14_on_update** present

## on_select
incomplete **payload** present

## init, confirm
missing type **FOLIO** in **person.creds**

## 12_on_update
**message_id** should be same as **update**'s **message_id**

## Flow Lumpsum_with_existing_folio
## select
**customer.creds** is not required in /select

swapnil-cyb commented:
@sourabhsaini11 @92shreyansh corrected the mentioned changes in additional purchase with existing folio and added logs in case of additional purchase existing folio
cc @Srimaan-Guntoju @arcybrilla

sourabhsaini11 commented:
@swapnil-cyb 
## 8b
## common_issue
**additional calls** select_2, on_select_2, 13_on_status, 14_on_update present

## on_select
incomplete **payload** present

## init, confirm
missing type **FOLIO** in person.creds

## 12_on_update
**message_id** should be same as update's **message_id**

## 8a
## common_issue
**additional calls** 11_on_status, 12_on_update present

## 10_on_update
**message_id** should be same as update's **message_id**
